exceptionCaught() event was fired

### Description
I am getting a lot of warnings like the below on Medalla right now. Using external signer and about 128keys assigned to this Teku instance.

**Expected behavior:** [What you expect to happen]
Not have any warnings

**Actual behavior:** [What actually happens]
getting a lot of warnings similar to 
```
12:03:46.868 INFO  - Completed regeneration of block 0x977115c2cc344909f205485e12766e4569fc09ce8ebecea86998b256a2ff5d7d at slot 546896 by replaying 5 blocks. Took 614ms
12:03:50.752 WARN  - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.channel.ChannelPipelineException: tech.pegasys.teku.networking.p2p.libp2p.rpc.RpcHandler$Controller.handlerRemoved() has thrown an exception.
        at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:640) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.DefaultChannelPipeline.destroyDown(DefaultChannelPipeline.java:876) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.DefaultChannelPipeline.destroyUp(DefaultChannelPipeline.java:844) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.DefaultChannelPipeline.destroy(DefaultChannelPipeline.java:836) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.DefaultChannelPipeline.access$700(DefaultChannelPipeline.java:46) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelUnregistered(DefaultChannelPipeline.java:1392) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:198) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:184) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelUnregistered(DefaultChannelPipeline.java:821) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:826) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:497) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.51.Final.jar:4.1.51.Final]
        at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: java.lang.NullPointerException
```
**Frequency:** [What percentage of the time does it occur?]
often

### Versions (Add all that apply)
* Software version:  teku/v0.12.11-dev-35d5b8a4/linux-x86_64/oracle_openjdk-java-14  in Docker

--- a/networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/rpc/RpcHandler.java
+++ b/networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/rpc/RpcHandler.java
@@ -29,7 +29,9 @@ import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.SimpleChannelInboundHandler;
 import java.time.Duration;
+import java.util.Optional;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes;
@@ -118,7 +120,7 @@ public class RpcHandler implements ProtocolBinding<Controller> {
   static class Controller extends SimpleChannelInboundHandler<ByteBuf> {
     private final NodeId nodeId;
     private final P2PChannel p2pChannel;
-    private RpcRequestHandler rpcRequestHandler;
+    private Optional<RpcRequestHandler> rpcRequestHandler = Optional.empty();
     private RpcStream rpcStream;
     private boolean readCompleted = false;
 
@@ -141,14 +143,14 @@ public class RpcHandler implements ProtocolBinding<Controller> {
 
     @Override
     protected void channelRead0(final ChannelHandlerContext ctx, final ByteBuf msg) {
-      rpcRequestHandler.processData(nodeId, rpcStream, msg);
+      runHandler(h -> h.processData(nodeId, rpcStream, msg));
     }
 
     public void setRequestHandler(RpcRequestHandler rpcRequestHandler) {
-      if (this.rpcRequestHandler != null) {
+      if (this.rpcRequestHandler.isPresent()) {
         throw new IllegalStateException("Attempt to set an already set data handler");
       }
-      this.rpcRequestHandler = rpcRequestHandler;
+      this.rpcRequestHandler = Optional.of(rpcRequestHandler);
 
       activeFuture.finish(
           () -> {
@@ -186,19 +188,23 @@ public class RpcHandler implements ProtocolBinding<Controller> {
     }
 
     private void onRemoteWriteClosed() {
-      if (!readCompleted && rpcRequestHandler != null) {
+      if (!readCompleted) {
         readCompleted = true;
-        rpcRequestHandler.readComplete(nodeId, rpcStream);
+        runHandler(h -> h.readComplete(nodeId, rpcStream));
       }
     }
 
     private void onChannelClosed() {
       try {
         onRemoteWriteClosed();
-        rpcRequestHandler.closed(nodeId, rpcStream);
+        runHandler(h -> h.closed(nodeId, rpcStream));
       } finally {
-        rpcRequestHandler = null;
+        rpcRequestHandler = Optional.empty();
       }
+    }
+
+    private void runHandler(final Consumer<RpcRequestHandler> action) {
+      rpcRequestHandler.ifPresentOrElse(action, this::closeAbruptly);
     }
 
     @VisibleForTesting

